backend/plonk/bn254/unmarshal.go

```diff
@@ -73,7 +71,7 @@ func UnmarshalSolidity(s []byte, nbCommits int) Proof {
 	// uint256[] selector_commit_api_at_zeta;
 	// uint256[] wire_committed_commitments;
 	for i := 0; i < nbCommits; i++ {
-		proof.BatchedProof.ClaimedValues[7+i].SetBytes(s[offset : offset+fr_size])
+		proof.BatchedProof.ClaimedValues[6+i].SetBytes(s[offset : offset+fr_size])
```

Comment:
if I'm not mistaken, `UnmarshalSolidity` is untested in our codebase --> can you add a simple round trip test with the `MarshalSolidity` func to ensure non regression / offset errors ?

Reply:
so I discussed a bit with OZ about that, in fact UnmarshalSolidity was used for debugging only (at the time there was an issue on testnet on I had to check the proof that was given by linea). At the time MarshalSolidity serialised all the elements of the proof. Now there has been an optimisation such that we don't serialise the opening of the linearised polynomial, it is recomputed in the verifier. To recompute this value we need the public inputs. So UnmarshalSolidity would have a weird signature (we would need to pass the public inputs). I told OZ that this function should be used internally for debugging only, I don't think we should keep it in the repo